### PR TITLE
feat/refactor: add cocoa FFI binding

### PR DESCRIPTION
**Pull request by vaind**
## :scroll: Description

Based on experiments in #1622 and as a prerequisite for #1611 - this adds Cocoa SDK FFI bindings generated by ffigen.
None of the bindings are actually used at all so this doesn't have any functional changes yet.

However, the following PRs can make use of FFI now. This is necessary for profiling because it needs to be started synchronously, which MethodChannels don't allow us to do.

#skip-changelog - this change should have no effect on library users at this time

## :pencil: Checklist

- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

## :crystal_ball: Next steps

- switch the profiling PR to use FFI so that we get correct samples
- consider switching from method channels to FFI (see #1444)

### CODE REVIEW

File: flutter/scripts/generate-cocoa-bindings.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+# Remove dependency on script exit (even in case of an error).
+trap "dart pub remove ffigen" EXIT
+
+# Currently we add the dependency only when the code needs to be generated because it depends
+# on Dart SDK 3.2.0 which isn't available on with Flutter stable yet.
+# Leaving the dependency in pubspec would block all contributors.
+# As for why this is coming from a fork - because we need a specific version of ffigen including PR 607 but not PR 601
+# which starts generating code not compatible with Dart SDK 2.17. The problem is they were merged in the wrong order...
+dart pub add 'dev:ffigen:{"git":{"url":"https://github.com/vaind/ffigen","ref":"4d714b0f2d11e9698a1885ec2d5de15f2ed827f1"}}'
```
Comment (bruno-garcia) [inline flutter/scripts/generate-cocoa-bindings.sh]: Should we fork this under `getsentry`? Or use straight up the `dart-lang` one?

```suggestion
dart pub add 'dev:ffigen:{"git":{"url":"https://github.com/getsentry/ffigen","ref":"4d714b0f2d11e9698a1885ec2d5de15f2ed827f1"}}'
```

**Review by bruno-garcia — CHANGES_REQUESTED** [review]

File: flutter/scripts/generate-cocoa-bindings.sh (hunk 1 above)
Comment (bruno-garcia) [reply flutter/scripts/generate-cocoa-bindings.sh]: Create a fork: https://github.com/getsentry/ffigen

@vaind will create a new branch with the sha we need, and we'll point to it from here

**Review by bruno-garcia — APPROVED** [review]
Please update the issue once the org in the repo is fixed here: https://github.com/getsentry/ffigen/issues/1

File: flutter/scripts/generate-cocoa-bindings.sh (hunk 1 above)
Comment (vaind) [reply flutter/scripts/generate-cocoa-bindings.sh]: done

File: .github/workflows/update-deps.yml [hunk 2]
```diff
@@ -3,11 +3,11 @@ name: Update Dependencies
...
   # Run every day.
   schedule:
-    - cron: '0 3 * * *'
+    - cron: "0 3 * * *"
   # And on on every PR merge so we get the updated dependencies ASAP, and to make sure the changelog doesn't conflict.
   push:
-    branches:
-      - main
+    # branches:
+    #   - main
```
Comment (krystofwoldrich) [inline .github/workflows/update-deps.yml]: Are the deps updates disabled for main on purpose? I don't see why would we want that?

Comment (vaind) [reply .github/workflows/update-deps.yml]: no, just realized this morning I also need to install Flutter SDK during the updater workflow so I have it enabled on all pushes now while working on it.

sorry for the confusion

**Pull request merged**